promql/query_logger.go

```diff
@@ -207,18 +213,20 @@ func (tracker ActiveQueryTracker) GetMaxConcurrent() int {
 
 func (tracker ActiveQueryTracker) Delete(insertIndex int) {
 	copy(tracker.mmapedFile[insertIndex:], strings.Repeat("\x00", entrySize))
+	tracker.mmapedFile.Flush()
 	tracker.getNextIndex <- insertIndex
 }
 
 func (tracker ActiveQueryTracker) Insert(ctx context.Context, query string) (int, error) {
 	select {
 	case i := <-tracker.getNextIndex:
-		fileBytes := tracker.mmapedFile
+		memMap := tracker.mmapedFile
```

Comment:
why do we need this variable at all?

Reply:
I can get rid of it.

I think it was meant to make the code more readable, but below doesn't look too bad either
```go
 copy(tracker.mmapedFile[start:], entry)
 copy(tracker.mmapedFile[start:], ",")
 tracker.mmapedFile.Flush()
```